src/services/external/excalidraw-backend/excalidraw.amqp.server.factory.provider.ts

```diff
@@ -0,0 +1,374 @@
+
+      logger?.verbose?.(
+        `User '${agentInfo.userID}' established connection`,
+        LogContext.EXCALIDRAW_SERVER
+      );
+
+      wsServer.to(socket.id).emit(INIT_ROOM);
+      // client events ONLY
+      socket.on(
+        JOIN_ROOM,
+        async (roomID: string) =>
+          await joinRoomEventHandler(
+            roomID,
+            agentInfo,
+            socket,
+            wsServer,
+            whiteboardRtService,
+            authorizationService,
+            excalidrawEventPublisher,
+            logger
+          )
+      );
+
+      socket.on(SERVER_BROADCAST, (roomID: string, data: ArrayBuffer) =>
+        serverBroadcastEventHandler(
+          roomID,
+          data,
+          socket,
+          excalidrawEventPublisher
+        )
+      );
+
+      socket.on(
+        SERVER_VOLATILE_BROADCAST,
+        (roomID: string, data: ArrayBuffer) =>
+          serverVolatileBroadcastEventHandler(
+            roomID,
+            data,
+            socket,
+            excalidrawEventPublisher
+          )
+      );
+
+      socket.on(
+        DISCONNECTING,
+        async () =>
+          await disconnectingEventHandler(
+            agentInfo,
+            wsServer,
+            socket,
+            logger,
+            excalidrawEventPublisher
+          )
+      );
+
+      socket.on(DISCONNECT, () =>
+        disconnectEventHandler(socket, excalidrawEventPublisher)
+      );
+    });
+  } catch (error) {
+    logger.error(error, LogContext.EXCALIDRAW_SERVER);
+  }
+};
+
+/* Sets the user into the context field or closes the connection */
+const authenticate = async (
+  kratosFrontEndApi: FrontendApi,
+  headers: Record<string, string | string[] | undefined>,
+  logger: LoggerService,
+  authService: AuthenticationService
+): Promise<AgentInfo> => {
+  const cookie = headers.cookie as string;
+
+  try {
+    const { data: session } = await kratosFrontEndApi.toSession({
+      cookie,
+    });
+
+    if (!session) {
+      logger.verbose?.('No Ory Kratos session', LogContext.EXCALIDRAW_SERVER);
+      return authService.createAgentInfo();
+    }
+
+    const oryIdentity = session.identity as OryDefaultIdentitySchema;
+    return authService.createAgentInfo(oryIdentity);
+  } catch (e: any) {
+    throw new Error(e?.message);
+  }
+};
+/* returns the user agent info */
+const getUserInfo = async (
+  kratosFrontEndApi: FrontendApi,
+  headers: Record<string, string | string[] | undefined>,
+  logger: LoggerService,
+  authService: AuthenticationService
+) => {
+  try {
+    return await authenticate(kratosFrontEndApi, headers, logger, authService);
+  } catch (e) {
+    const err = e as Error;
+    logger.error(
+      `Error when trying to authenticate with excalidraw server: ${err.message}`,
+      err.stack,
+      LogContext.EXCALIDRAW_SERVER
+    );
+    return undefined;
+  }
+};
+/* This event is coming from the client; whenever they request to join a room */
+const joinRoomEventHandler = async (
```

Comment:
Having the event handlers in a separate files will make the code more readable. Generally, it's a good practice to have files with < 200 lines as well.